jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh

```diff
@@ -99,7 +99,8 @@ log 'Installing prerequisites...'
 retry 5 curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
 retry 5 apt-key update
 
-retry 5 apt-get update
+#TODO: Remove this flag once we migrate to debian11
```

Comment:
If there's a related ticket (for Dataproc version update?) that will allow us to migrate to Debian 11, could you add the link to this ticket, and also add a comment to that ticket to consider doing the migration here? I have a feeling, it'll be too easy to forget about it otherwise :)

Reply:
I can't seem to find it. Should I create it?

Reply:
Just added the note

> If, at the time of implementing this ticket, the list of Dataproc-supported Debian images includes Debian 11 or higher, it would be good to upgrade the Debian base image for the leonardo-custom-image Jenkins job.

to the description of [Support latest hail image](https://broadworkbench.atlassian.net/browse/IA-2924)  😊